### PR TITLE
vSphere UPI on VCM

### PR DESCRIPTION
**Pull request by rvanderp3**

### CODE REVIEW

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-upi-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-upi-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-upi-zones-f28 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-vsphere-ipi-zones-f28 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-externallb-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-upi-zones

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones

4.15 zones didn't move

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi-zones

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

**Comment by jcpowermac** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-release-4.16-platform-none-vsphere-e2e-operator

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-upi periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-upi

File: ci-operator/step-registry/cucushift/agent/vsphere/conf/vips/cucushift-agent-vsphere-conf-vips-commands.sh [hunk 1]
```diff
@@ -17,6 +17,10 @@ fi
 declare vlanid
 declare primaryrouterhostname
 source "${SHARED_DIR}/vsphere_context.sh"
+# These two environment variables are coming from vsphere_context.sh and
+# the file they are assigned to is not available in this step.
+unset SSL_CERT_FILE
+unset GOVC_TLS_CA_CERTS
```
Comment (vr4manta) [inline ci-operator/step-registry/cucushift/agent/vsphere/conf/vips/cucushift-agent-vsphere-conf-vips-commands.sh:24]: Are these needed?  They are executed right after this.

File: ci-operator/step-registry/cucushift/agent/vsphere/deprovision/cucushift-agent-vsphere-deprovision-commands.sh [hunk 2]
```diff
@@ -14,6 +14,8 @@ source "${SHARED_DIR}/govc.sh"
 
 declare vsphere_portgroup
 source "${SHARED_DIR}/vsphere_context.sh"
+unset SSL_CERT_FILE
+unset GOVC_TLS_CA_CERTS
```
Comment (vr4manta) [inline ci-operator/step-registry/cucushift/agent/vsphere/deprovision/cucushift-agent-vsphere-deprovision-commands.sh:19]: Are these needed?  They are executed right after this.

File: ci-operator/step-registry/vsphere/deprovision/bastionhost/vsphere-deprovision-bastionhost-commands.sh [hunk 3]
```diff
@@ -8,6 +8,8 @@ trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wa
 
 export AWS_SHARED_CREDENTIALS_FILE=/var/run/vault/vsphere-aws/.awscred
 source "${SHARED_DIR}/govc.sh"
+unset SSL_CERT_FILE
+unset GOVC_TLS_CA_CERTS
```
Comment (vr4manta) [inline ci-operator/step-registry/vsphere/deprovision/bastionhost/vsphere-deprovision-bastionhost-commands.sh:13]: Are these needed?  They are executed right after this.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-8-upi-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-stable-vsphere-ipi-zones-f14

**Comment by jianlinliu** [comment]
@rvanderp3 @jcpowermac per the [log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/52636/rehearse-52636-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-8-upi-f28/1810860935615614976), upi-conf-vsphere-ova-vcm step failed.

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-8-upi-f28

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-8-upi-f28

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-8-upi-f28

**Comment by sebsoto** [comment]
/approve

No issues with this as long as the windows jobs remain functional

**Pull request merged**